Lushu/src/main/kotlin/lushu/ContextGrammar/Grammar/Parser.kt

```diff
@@ -0,0 +1,12 @@
+package lushu.ContextGrammar.Grammar
+
+class Parser(private val rules: Rules = Rules()) {
+
+    fun parsing(wordsList: MutableList<String>): List<String> {
```

Comment:
```suggestion
    fun parse(wordsList: List<String>): List<String> {
```
Funcitons names should be verbs. Also, please use an imutable list instead of a mutable list. Note that this may require changing other code too

Reply:
Also, could you please add an explanation on why this function needs to exist at all? I mean, it just forwards the parameters to the `rules` object.